{bio}[GCCcore/12.3.0] NextDenovo v2.5.2

pavelToman opened the pull request:
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/397

pavelToman commented:
@boegelbot please test @ jsc-zen3

pavelToman commented:
@boegelbot please test @ generoso

boegel commented:
@pavelToman For the failing test report on `generoso`, it looks like there's some linking trouble:
```
/project/boegelbot/Rocky8/haswell/software/binutils/2.40-GCCcore-12.3.0/bin/ld: ../lib/htslib//libhts.a(hfile_libcurl.o): in function `easy_errno':
hfile_libcurl.c:(.text+0x5d): undefined reference to `curl_easy_getinfo'
/project/boegelbot/Rocky8/haswell/software/binutils/2.40-GCCcore-12.3.0/bin/ld: hfile_libcurl.c:(.text+0xb5): undefined reference to `curl_easy_getinfo'
/project/boegelbot/Rocky8/haswell/software/binutils/2.40-GCCcore-12.3.0/bin/ld: ../lib/htslib//libhts.a(hfile_libcurl.o): in function `wait_perform':
hfile_libcurl.c:(.text+0x5a5): undefined reference to `curl_multi_fdset'
```

Is it trying to build it's own copy of `HTSlib`, rather than using a dependency? Can we avoid that?

pavelToman commented:
> @pavelToman For the failing test report on `generoso`, it looks like there's some linking trouble:
> 
> ```
> /project/boegelbot/Rocky8/haswell/software/binutils/2.40-GCCcore-12.3.0/bin/ld: ../lib/htslib//libhts.a(hfile_libcurl.o): in function `easy_errno':
> hfile_libcurl.c:(.text+0x5d): undefined reference to `curl_easy_getinfo'
> /project/boegelbot/Rocky8/haswell/software/binutils/2.40-GCCcore-12.3.0/bin/ld: hfile_libcurl.c:(.text+0xb5): undefined reference to `curl_easy_getinfo'
> /project/boegelbot/Rocky8/haswell/software/binutils/2.40-GCCcore-12.3.0/bin/ld: ../lib/htslib//libhts.a(hfile_libcurl.o): in function `wait_perform':
> hfile_libcurl.c:(.text+0x5a5): undefined reference to `curl_multi_fdset'
> ```
> 
> Is it trying to build it's own copy of `HTSlib`, rather than using a dependency? Can we avoid that?

There is some weird problem - as soon as I added HTSlib as a dep (not change anything else) it start failing on make step -> https://github.com/vscentrum/vsc-software-stack/blob/wip/397_NextDenovo/log1.txt

I found that there is no problem with HTSlib but with its dep cURL - as soon as I added cURL as a dep of NextDenovo - it failing with the same errors -> https://github.com/vscentrum/vsc-software-stack/blob/wip/397_NextDenovo/log2.txt

So I look what version of curl is on donphan - curl 7.61.1, and created cURL for GCCcore/12.3.0 with version 7.86.0, 7.66.0, 7.60.0 - still there is the same error during make step -> https://github.com/vscentrum/vsc-software-stack/blob/wip/397_NextDenovo/log3.txt

Seems there is something else going on

pavelToman commented:
Ah, found something - this few last commits [here](https://github.com/Nextomics/NextDenovo/commits/1580f95d16d2f6886a773c89a83031127e3b0900/) looks like they can fix this issue.
I've tried to build NextDenovo from master (with these commits) and now it works only when cURL is in deps.

pavelToman commented:
@boegelbot please test @ generoso

pavelToman commented:
@boegelbot please test @ generoso

pavelToman commented:
@boegelbot please test @ jsc-zen3

boegel commented on easybuild/easyconfigs/n/NextDenovo/NextDenovo-2.5.2-20240510-GCCcore-12.3.0.eb:
```diff
@@ -0,0 +1,54 @@
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+source_urls = ['https://github.com/Nextomics/NextDenovo/archive/']
+sources = [{'download_filename': '0e5fa5d.tar.gz', 'filename': SOURCE_TAR_GZ}]
```
use `local_commit` here:
```suggestion
sources = [{'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ}]
```

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3124.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.9.18
See https://gist.github.com/boegel/e4a7859a3f1c12d8f1a45cb6637ad6e0 for a full test report.

boegel commented:
Going in, thanks @pavelToman!
